messaging/broker.go

```diff
@@ -1285,8 +1297,14 @@ func (dec *MessageDecoder) Decode(m *Message) error {
 	m.unmarshalHeader(b[:])
 
 	// Read data.
-	if _, err := io.ReadFull(dec.r, m.Data); err != nil {
-		return fmt.Errorf("read body: %s", err)
+	if n, err := io.ReadFull(dec.r, m.Data); err == io.EOF || err == io.ErrUnexpectedEOF {
+		if _, err := dec.r.Seek(-int64(MessageHeaderSize+n), os.SEEK_CUR); err != nil {
+			return fmt.Errorf("cannot unread header+data: n=%d, err=%s", n, err)
+		}
+		warnf("unexpected eof(1): len=%d, n=%d, err=%s", len(m.Data), n, err)
```

Comment:
Yeah, for the short term. It'll give us a little extra data. We don't have these variables returned to the caller and there's no logger in here so I couldn't think of a better way to inject it. We can remove the warns in the future.